tests/model_serving/model_server/maas_billing/utils.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+from typing import Dict, Optional
+
+import base64
+import requests
+from ocp_resources.gateway_gateway_networking_k8s_io import Gateway
+from ocp_resources.ingress_config_openshift_io import Ingress as IngressConfig
+from requests import Response
+
+
+def host_from_ingress_domain(client) -> str:
+    """Return 'maas.<ingress-domain>'"""
+    ingress_config = IngressConfig(name="cluster", client=client, ensure_exists=True)
+    domain = ingress_config.instance.spec.get("domain")
+    assert domain, "Ingress 'cluster' missing spec.domain (ingresses.config.openshift.io)"
+    return f"maas.{domain}"
+
+
+def scheme_from_gateway(gw: Gateway) -> str:
+    """
+    Decide 'http' or 'https' from Gateway listeners.
+    Rule: if any listener has protocol HTTPS or tls set, return 'https'; else 'http'.
+    listeners is a list[dict] per the Gateway wrapper.
+    """
+    listeners = gw.instance.spec.get("listeners", [])
+    for listener in listeners:
+        protocol = (listener.get("protocol") or "").upper()
+        if protocol == "HTTPS" or listener.get("tls"):
```

Comment:
I see that even when protocol is  `TLS`, then tls would be set:
```
tls	<Object>
    TLS is the TLS configuration for the Listener. This field is required if
    the Protocol field is "HTTPS" or "TLS". It is invalid to set this field
    if the Protocol field is "HTTP", "TCP", or "UDP".
```
In that case do you want to return https?